### DISCUSSION TITLE
ref引用类的使用问题

### DISCUSSION THREAD

**Discussion by ChrisLo0751** [opentiny/tiny-engine#155] [General]
当我为组件绑定了ref值为theGrid
![51802557906bdd6c82c914c07d6455c](https://github.com/opentiny/tiny-engine/assets/73991742/c8db0b9c-b786-453e-89f5-28af179b575b)

接下来，期望通过 `this.$refs.theGrid` 获取该组件，结果获取失败，我尝试打印 `this.$refs`显示结果为undefined

```
function onEdit(event) {
  console.log(this.$refs)
}


// 结果
 undefined
```

按照正常道理，refs 是一个实例会被挂载在vue上，于是我打印了vue这个实例查看,结果发现貌似ref的值报错了
![1702605808760](https://github.com/opentiny/tiny-engine/assets/73991742/5d78b77f-9d23-49f9-8b9b-f5e857714e15)

请帮我勘误，是我的使用调试错误，还是本身功能有bug。谢谢

**Comment by ChrisLo0751** [comment]
@chilingling

**Comment by chilingling** [comment]
lowcode 获取实例方式：`this.$('输入框填入的 ref 名称')`，比如你第一张截图定义的 ref 可以这样获取： `this.$('theGrid')` 获取到 ref 实例

**Reply by ChrisLo0751** [reply]
```javascript
console.log(this.$('theGrid'))
````
我尝试打印该实例的信息时，收到报错：Cannot convert object to primitive value

**Reply by chilingling** [reply]
参考 #144 ，实测是没问题的。或者麻烦给出详细的配置信息，以及报错过程

**Reply by ChrisLo0751** [reply]
经本人重复排查，确定是表格组件有bug。

必现步骤如下：

1、拖拽表格组件进入画布，并绑定ref值：tableRef

2、拖拽按钮，并绑定ref值：buttonRef

3、为按钮增加点击事件，代码如下：
```javascript
function onClickNew(event) {
  console.log(this.$('buttonRef'))
  console.log(this.$('tableRef'))
}
````

4、查看控制台输出

![图片](https://github.com/opentiny/tiny-engine/assets/73991742/5c87c632-0765-444d-984f-fcfbcb189ea2)

如图可以看到，button的实例被打印出来，而表格组件的输出是：about:srcdoc:160  TypeError: Cannot convert object to primitive value
